### PR TITLE
Add ISO 8601 duration to feed stats endpoint

### PR DESCRIPTION
**Pull request by dragospitica**

### CODE REVIEW

**Comment by AdamDM** [comment]
lgtm

File: src/main/java/org/atlasapi/query/v2/FeedStatsController.java [hunk 1]
```diff
@@ -52,8 +54,24 @@ public FeedStatsController(
...
 
     @RequestMapping(value="/3.0/feeds/youview/{publisher}/statistics.json", method = RequestMethod.GET)
-    public void statistics(HttpServletRequest request, HttpServletResponse response,
-            @PathVariable("publisher") String publisherStr) throws IOException {
+    public void statistics(
+            HttpServletRequest request,
+            HttpServletResponse response,
+            @PathVariable("publisher") String publisherStr
+    ) throws IOException {
+        String isoDuration = request.getParameter("duration");
```
Comment (emilssolmanis) [inline src/main/java/org/atlasapi/query/v2/FeedStatsController.java]: This is not necessary, add it as an annotated `@RequestParam`.

File: src/main/java/org/atlasapi/query/v2/FeedStatsController.java [hunk 2]
```diff
@@ -52,8 +54,24 @@ public FeedStatsController(
...
+
+        if (Strings.isNullOrEmpty(isoDuration)) {
```
Comment (emilssolmanis) [inline src/main/java/org/atlasapi/query/v2/FeedStatsController.java]: This is going to break the endpoint for existing clients of the API, you probably need to figure out  a default value. If not, just use `@RequestParam(required=true)`.

File: src/main/java/org/atlasapi/query/v2/FeedStatsController.java [hunk 3]
```diff
@@ -52,8 +54,24 @@ public FeedStatsController(
...
+            errorViewFor(request,
+                    response,
+                    AtlasErrorSummary.forException(new IllegalArgumentException(
+                            "The fromTime request parameter must be specified"
+                    ))
+            );
+        }
+
+        java.time.Duration duration = java.time.Duration.parse(isoDuration);
```
Comment (emilssolmanis) [inline src/main/java/org/atlasapi/query/v2/FeedStatsController.java]: Joda isn't used in this class, no need for absolute qualifier. Just import it.

Comment (emilssolmanis) [inline src/main/java/org/atlasapi/query/v2/FeedStatsController.java]: The name is off, `duration` has no semantic value, it's literally the type name of the variable.

File: src/main/java/org/atlasapi/query/v2/FeedStatsController.java [hunk 4]
```diff
@@ -52,8 +55,15 @@ public FeedStatsController(
     }
 
     @RequestMapping(value="/3.0/feeds/youview/{publisher}/statistics.json", method = RequestMethod.GET)
-    public void statistics(HttpServletRequest request, HttpServletResponse response,
-            @PathVariable("publisher") String publisherStr) throws IOException {
+    public void statistics(
+            HttpServletRequest request,
+            HttpServletResponse response,
+            @PathVariable("publisher") String publisherStr,
+            @RequestParam(value = "isoDuration") String isoDuration
```
Comment (emilssolmanis) [inline src/main/java/org/atlasapi/query/v2/FeedStatsController.java]: `value` is the default and is optional, you can just write `@RequestParam("isoDuration")`, but again, that is a very non descriptive argument name.

File: src/main/java/org/atlasapi/query/v2/FeedStatsController.java [hunk 5]
```diff
@@ -52,8 +55,15 @@ public FeedStatsController(
...
+    ) throws IOException {
+        // we parse the ISO 8601 duration as a period because hours are inexact in terms of milliseconds
+        Period requiredPeriod = Period.parse(isoDuration);
```
Comment (emilssolmanis) [inline src/main/java/org/atlasapi/query/v2/FeedStatsController.java]: More bad naming.

**Pull request merged**